internal/kafka/source_result.go

```diff
@@ -14,18 +12,12 @@ const (
 )
 
 type SourceResult struct {
-	ResourceID string `json:"resource_id"`
-
-	// Resource type of the source
-	ResourceType string `json:"resource_type"`
-
-	Status StatusType `json:"status"`
-
-	Err error `json:"error"`
-
-	Identity identity.Principal `json:"-"`
-
-	MissingPermissions []string `json:"-"`
+	MessageContext     context.Context `json:"-"` // Carries logger and identity
+	ResourceID         string          `json:"resource_id"`
+	ResourceType       string          `json:"resource_type"`
+	Status             StatusType      `json:"status"`
+	Err                error           `json:"-"` // Sources do not support error field
+	MissingPermissions []string        `json:"-"` // Sources do not support reason field
 }
```

Comment:
Before the patch, we were serializing "err" field, this changed it. Sources does not support this field, so let’s not send it.